app/services/provider_setup.rb

```diff
@@ -23,11 +23,21 @@ def next_agreement_pending
   end
 
   def next_relationship_pending
-    permissions = ProviderRelationshipPermissions.find_by(
+    permissions = ProviderRelationshipPermissions.order(:created_at).find_by(
       setup_at: nil,
       training_provider: @provider_user.providers,
     )
 
+    if permissions.blank?
+      setup_permissions = ProviderRelationshipPermissions
+        .includes(:training_provider, :ratifying_provider)
+        .where(training_provider: @provider_user.providers)
+        .where.not(setup_at: nil)
+        .order(:created_at)
+
+      permissions = setup_permissions.reject(&:valid?)&.first
+    end
```

Comment:
Isn't the logic here 

- get all the permissions on which I'm the training provider
- if any lack setup_at, we need to set them up
- if any are invalid, we need to set them up

so rather than two passes we could do something like

```
ProviderRelationshipPermissions
  .where(training_provider: @provider_user.providers)
  .select { |prp| prp.setup_at.blank? || prp.invalid? }
```